### ISSUE TITLE
How to constuct train and test splits of the LiveJournal dataset?

### ISSUE THREAD

**Issue by JessicaAlphy** [facebookresearch/PyTorch-BigGraph#97]
Hello, the raw dataset of LiveJournal has two columns, like the following style:
```
#Directed graph( each unordered pair of nodes is saved once): soc-LiveJournal11.txt
#Directed LiveJournal friendship social network
#Nodes: 4847571   Edges: 68993773
#FromNodeId   ToNodeId
0    1
0    2
0    3
0    4
0    5
0    6
0    7
0    8
```
My question is 1. In the file `PyTorch-BigGraph/torchbiggraph/examples/livejournal.py` ,

the raw LiveJournal dataset has two columns, do we need to preprocess it to change the two-column dataset into three columns?  like FB15K dataset. If we need to change, how to do?

```
FILENAMES = {
    'train': 'train.txt',
    'test': 'test.txt',
}
TRAIN_FRACTION = 0.75
```

2. You say that construct the train and test dataset by 0.75 and 0.25 split, just the `|total lines| * 0.75 `as the train dataset, and `|total lines|*0.25` as the test dataset? 

3.The command of executing training of the LiveJournal dataset? 

Looking forward to your reply.
Best regards.
Jessica Cheshire

**Comment by lerks** [comment]
1. In `import_from_tsv`, the `--rel-col` command-line flag is optional. If it's not given, then the relation type for an edge is just assumed to be 0, without it being looked up anywhere in the file. In that case the TSV file only needs to contain two columns, for the left- and right-hand side entities.
2. Yes, the train set contains `0.75 * |total lines|` (rounded down to the nearest integer), whereas the test set contains everything else. The lines are shuffled before being split.
3. Training is launched in exactly the same way as with FB15k, except that you need to use the [`torchbiggraph/examples/configs/livejournal_config.py` file](https://github.com/facebookresearch/PyTorch-BigGraph/blob/master/torchbiggraph/examples/configs/livejournal_config.py).

**Comment by lerks** [comment]
Closing as there's been no follow-up in 10 days. Please reopen if needed.
